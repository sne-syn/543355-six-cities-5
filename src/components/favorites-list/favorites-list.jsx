import React from 'react';
import PropTypes from 'prop-types';
import CardFavorites from '../card/card-favorites/card-favorites';
import LocationsItem from '../locations-item/locations-item';

const FavoritesList = ({favoritesOffersOnly}) => {
  const someOffers = favoritesOffersOnly.filter(((offer) => {
    return offer.city === `Amsterdam`;
  }));

  return (
    <ul className="favorites__list">
      <li className="favorites__locations-items">
        <div className="favorites__locations locations locations--current">
          <LocationsItem cityName={favoritesOffersOnly[1].city} />
        </div>
        <div className="favorites__places">
          {someOffers.map((offer) => {
            return (
              <CardFavorites
                key={offer.id}
                offer={offer}
              />
            );
          })}
        </div>
      </li>
    </ul>
  );
};

FavoritesList.propTypes = {
  favoritesOffersOnly: PropTypes.array.isRequired
};

export default FavoritesList;
