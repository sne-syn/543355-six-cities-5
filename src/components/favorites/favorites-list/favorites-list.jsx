import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {changeActiveElement} from '../../../store/action';
import {getActiveElement, getOffers} from '../../../store/offers-data/offers-data-selectors';
import FavoritesCardsList from '../favorites-cards-list/favorites-cards-list';
import LocationsItem from '../../locations-item/locations-item';

const arrangeFavoritesByCity = (favoritesOffers) => {
  let collection = new Set();
  for (let value of favoritesOffers) {
    collection.add(value.city.name);
  }
  let citiesWithFavoritesOffers = Array.from(collection);
  return citiesWithFavoritesOffers;
};

const FavoritesList = ({favoritesOffers, changeLocation}) => {
  const citiesWithFavoritesOffers = arrangeFavoritesByCity(favoritesOffers);
  return (
    <ul className="favorites__list">
      {citiesWithFavoritesOffers.map((city, i) => {
        return (
          <li key={`city-${i}`} className="favorites__locations-items">
            <div className="favorites__locations locations locations--current">
              <div className="locations__item">
                <LocationsItem cityName={city} onClick={changeLocation}/>
              </div>
            </div>
            <FavoritesCardsList city={city} offers={favoritesOffers}/>
          </li>
        );
      })
      }
    </ul>
  );
};

FavoritesList.propTypes = {
  favoritesOffers: PropTypes.array.isRequired,
  activeElement: PropTypes.string.isRequired,
  changeLocation: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    favoritesOffers: getOffers(state),
    activeElement: getActiveElement(state),
  };
}

const mapDispatchToProps = (dispatch) => ({
  changeLocation(evt) {
    dispatch(changeActiveElement(evt));
  }
});

export {FavoritesList};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesList);
