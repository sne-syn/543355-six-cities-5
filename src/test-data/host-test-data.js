export const hostPro = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: true,
  name: `Janette`,
};
export const host = {
  avatar: `https://robohash.org/61?set=set2&size=74x74`,
  id: 5,
  isPro: false,
  name: `Janette`,
};
