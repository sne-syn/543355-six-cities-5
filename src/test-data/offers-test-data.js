export const offers = [{
  bedrooms: 1,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Paris`,
  },
  description: `Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish`,
  goods: [`cabel TV`, `heating`, `kitchen`, `towels`, `washing machine`, `washer`, `dishwasher`, `conditioner`, `elevator`, `coffee machine`],
  host: {
    avatar: `https://robohash.org/28?set=set2&size=74x74`,
    id: 4,
    isPro: false,
    name: `Salom`,
  },
  id: 1,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/34LWdhj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/33Qpecj`, `https://bit.ly/30YVQif`, `https://bit.ly/36TgIv7`, `https://bit.ly/3lEt2DG`],
  isFavorite: true,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 8,
  price: 207,
  rating: 1.4018800815887484,
  title: `La Latina- Small Charming Studio`,
  type: `room`,
}, {
  bedrooms: 3,
  city: {
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10,
    },
    name: `Brussels`,
  },
  description: `In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2`,
  goods: [`elevator`, `dishwasher`, `coffee machine`, `washer`, `washing machine`],
  host: {
    avatar: `https://robohash.org/82?set=set2&size=74x74`,
    id: 6,
    isPro: false,
    name: `Whitney`,
  },
  id: 2,
  previewImage: `https://bit.ly/34LWdhj`,
  images: [`https://bit.ly/30YVQif`, `https://bit.ly/3lEt2DG`, `https://bit.ly/36TgIv7`, `https://bit.ly/33Qpecj`, `https://bit.ly/36Ys6Wn`, `https://bit.ly/34LWdhj`],
  isFavorite: false,
  isPremium: false,
  location: {latitude: 48.865610000000004,
    longitude: 2.350499,
    zoom: 5},
  maxGuests: 5,
  price: 194,
  rating: 1.9069547983791004,
  title: `Estudio/suite completo en el centro de Segovia`,
  type: `house`,
}];
