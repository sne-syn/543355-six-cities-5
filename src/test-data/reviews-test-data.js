export const review = {
  author: `Beaulah`,
  avatar: `https://robohash.org/54?set=set2&size=54x54`,
  date: `2004-09-06T16:12:32.554Z`,
  id: 0,
  rating: 1,
  comment: `Antoni is a great host and us a wonderful home. His suggestions for places to go were killer!10/10 would stay again.`,
};

export const reviews = [{
  author: `Ione`,
  avatar: `https://robohash.org/98?set=set2&size=54x54`,
  date: `2007-10-13T16:12:32.554Z`,
  id: 1,
  rating: 4,
  comment: `There’s a stunning beach with great snorkelling just a 6min drive, a few other beautiful beaches close by too.`,
}, {
  author: `Bruno`,
  avatar: `https://robohash.org/53?set=set2&size=54x54`,
  date: `2005-08-25T16:12:32.554Z`,
  id: 2,
  rating: 3,
  comment: `Maria’s place is at a perfect location, very easy to get around with public transportation and walking distance to awesome bars and restaurants. Her place is great, very clean and has everything you need.`,
}];
