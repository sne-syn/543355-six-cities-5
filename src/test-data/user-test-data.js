export const user = {
  userAvatar: `https://robohash.org/82?set=set2&size=74x74`,
  userEmail: `trust_no_1@gmail.com`,
  userId: 1,
  userIsPro: false,
  userName: `fox_mulder`
};
