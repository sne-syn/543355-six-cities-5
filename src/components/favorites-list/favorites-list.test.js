import React from 'react';
import renderer from 'react-test-renderer';
import {FavoritesList} from './favorites-list';
import {offers} from '../../test-data/offers-test-data';

jest.mock(`../favorites-cards-list/favorites-cards-list`, () => `FavoritesCardsList`);
jest.mock(`../locations-item/locations-item`, () => `LocationsItem`);

const noop = () => {};

describe(`<FavoritesList /> test`, () => {
  it(` should render FavoritesList correctly`, () => {
    const tree = renderer.create((
      <FavoritesList
        activeElement={`Paris`}
        favorites={[offers[0]]}
        changeLocation={noop}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`should render FavoritesList correctly with multiple favorites`, () => {
    const tree = renderer.create((
      <FavoritesList
        activeElement={`Amsterdam`}
        favorites={offers}
        changeLocation={noop}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
