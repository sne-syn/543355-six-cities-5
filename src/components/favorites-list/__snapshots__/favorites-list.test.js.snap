// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoritesList /> test  should render FavoritesList correctly 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <LocationsItem
          cityName="Paris"
          onClick={[Function]}
        />
      </div>
    </div>
    <FavoritesCardsList
      city="Paris"
      offers={
        Array [
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Paris",
            },
            "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
            "goods": Array [
              "cabel TV",
              "heating",
              "kitchen",
              "towels",
              "washing machine",
              "washer",
              "dishwasher",
              "conditioner",
              "elevator",
              "coffee machine",
            ],
            "host": Object {
              "avatar": "https://robohash.org/28?set=set2&size=74x74",
              "id": 4,
              "isPro": false,
              "name": "Salom",
            },
            "id": 1,
            "images": Array [
              "https://bit.ly/34LWdhj",
              "https://bit.ly/36Ys6Wn",
              "https://bit.ly/33Qpecj",
              "https://bit.ly/30YVQif",
              "https://bit.ly/36TgIv7",
              "https://bit.ly/3lEt2DG",
            ],
            "isFavorite": true,
            "isPremium": false,
            "location": Object {
              "latitude": 48.865610000000004,
              "longitude": 2.350499,
              "zoom": 5,
            },
            "maxGuests": 8,
            "previewImage": "https://bit.ly/34LWdhj",
            "price": 207,
            "rating": 1.4018800815887484,
            "title": "La Latina- Small Charming Studio",
            "type": "room",
          },
        ]
      }
    />
  </li>
</ul>
`;

exports[`<FavoritesList /> test should render FavoritesList correctly with multiple favorites 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <LocationsItem
          cityName="Paris"
          onClick={[Function]}
        />
      </div>
    </div>
    <FavoritesCardsList
      city="Paris"
      offers={
        Array [
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Paris",
            },
            "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
            "goods": Array [
              "cabel TV",
              "heating",
              "kitchen",
              "towels",
              "washing machine",
              "washer",
              "dishwasher",
              "conditioner",
              "elevator",
              "coffee machine",
            ],
            "host": Object {
              "avatar": "https://robohash.org/28?set=set2&size=74x74",
              "id": 4,
              "isPro": false,
              "name": "Salom",
            },
            "id": 1,
            "images": Array [
              "https://bit.ly/34LWdhj",
              "https://bit.ly/36Ys6Wn",
              "https://bit.ly/33Qpecj",
              "https://bit.ly/30YVQif",
              "https://bit.ly/36TgIv7",
              "https://bit.ly/3lEt2DG",
            ],
            "isFavorite": true,
            "isPremium": false,
            "location": Object {
              "latitude": 48.865610000000004,
              "longitude": 2.350499,
              "zoom": 5,
            },
            "maxGuests": 8,
            "previewImage": "https://bit.ly/34LWdhj",
            "price": 207,
            "rating": 1.4018800815887484,
            "title": "La Latina- Small Charming Studio",
            "type": "room",
          },
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Brussels",
            },
            "description": "In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2",
            "goods": Array [
              "elevator",
              "dishwasher",
              "coffee machine",
              "washer",
              "washing machine",
            ],
            "host": Object {
              "avatar": "https://robohash.org/82?set=set2&size=74x74",
              "id": 6,
              "isPro": false,
              "name": "Whitney",
            },
            "id": 2,
            "images": Array [
              "https://bit.ly/30YVQif",
              "https://bit.ly/3lEt2DG",
              "https://bit.ly/36TgIv7",
              "https://bit.ly/33Qpecj",
              "https://bit.ly/36Ys6Wn",
              "https://bit.ly/34LWdhj",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 48.865610000000004,
              "longitude": 2.350499,
              "zoom": 5,
            },
            "maxGuests": 5,
            "previewImage": "https://bit.ly/34LWdhj",
            "price": 194,
            "rating": 1.9069547983791004,
            "title": "Estudio/suite completo en el centro de Segovia",
            "type": "house",
          },
        ]
      }
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <LocationsItem
          cityName="Brussels"
          onClick={[Function]}
        />
      </div>
    </div>
    <FavoritesCardsList
      city="Brussels"
      offers={
        Array [
          Object {
            "bedrooms": 1,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Paris",
            },
            "description": "Gran suite con entrada independiente y baño interior privado en el centro de Segovia. Privacidad y comodidad a tan sólo tres minutos del acueducto.Amplia y luminosa, con 2 camas de 105 (+ 1 opcional de 90), baño completo, calefacción, mininevera, microondas, vajilla, cubertería básica, Internet y un balcón grande y soleado.I speak English, Italian and Spanish",
            "goods": Array [
              "cabel TV",
              "heating",
              "kitchen",
              "towels",
              "washing machine",
              "washer",
              "dishwasher",
              "conditioner",
              "elevator",
              "coffee machine",
            ],
            "host": Object {
              "avatar": "https://robohash.org/28?set=set2&size=74x74",
              "id": 4,
              "isPro": false,
              "name": "Salom",
            },
            "id": 1,
            "images": Array [
              "https://bit.ly/34LWdhj",
              "https://bit.ly/36Ys6Wn",
              "https://bit.ly/33Qpecj",
              "https://bit.ly/30YVQif",
              "https://bit.ly/36TgIv7",
              "https://bit.ly/3lEt2DG",
            ],
            "isFavorite": true,
            "isPremium": false,
            "location": Object {
              "latitude": 48.865610000000004,
              "longitude": 2.350499,
              "zoom": 5,
            },
            "maxGuests": 8,
            "previewImage": "https://bit.ly/34LWdhj",
            "price": 207,
            "rating": 1.4018800815887484,
            "title": "La Latina- Small Charming Studio",
            "type": "room",
          },
          Object {
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.370216,
                "longitude": 4.895168,
                "zoom": 10,
              },
              "name": "Brussels",
            },
            "description": "In full nature and very close to public transport. Green area surrounded by all kinds of sports facilities including a golf course, paddle, artificial grass soccer, pelota court. In an urbanization of villas with landscaped plots of about 500 m2",
            "goods": Array [
              "elevator",
              "dishwasher",
              "coffee machine",
              "washer",
              "washing machine",
            ],
            "host": Object {
              "avatar": "https://robohash.org/82?set=set2&size=74x74",
              "id": 6,
              "isPro": false,
              "name": "Whitney",
            },
            "id": 2,
            "images": Array [
              "https://bit.ly/30YVQif",
              "https://bit.ly/3lEt2DG",
              "https://bit.ly/36TgIv7",
              "https://bit.ly/33Qpecj",
              "https://bit.ly/36Ys6Wn",
              "https://bit.ly/34LWdhj",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 48.865610000000004,
              "longitude": 2.350499,
              "zoom": 5,
            },
            "maxGuests": 5,
            "previewImage": "https://bit.ly/34LWdhj",
            "price": 194,
            "rating": 1.9069547983791004,
            "title": "Estudio/suite completo en el centro de Segovia",
            "type": "house",
          },
        ]
      }
    />
  </li>
</ul>
`;
